File: invertirText.py
cadena = "Hola Mundo"
cadena_invertidaSlicing = cadena[::-1]
print(cadena_invertidaSlicing)  # Resultado: "odnuM aloH"

cadena_invertidaFor = ""
for caracter in cadena[::-1]: # [start:stop:step]
    cadena_invertidaFor += caracter
print(cadena_invertidaFor)  # Resultado: "odnuM aloH"


cadena_invertidaM = ''.join(reversed(cadena))
print(cadena_invertidaM)  # Resultado: "odnuM aloH"


# "rebanado inverso" o "slicing inverso".
# Técnica que te permite extraer parte de una secuencia (como una cadena de texto, lista o tupla) utilizando una notación especial [start:stop:step].
# start: Índice de inicio de la rebanada (inclusive).
# stop: Índice de fin de la rebanada (exclusivo).
# step: Tamaño del paso para la rebanada (puede ser negativo para rebanar en reversa).

File: invertirText.js
let cadena = "Hola Mundo";

let textInvertMethod = cadena.split("").reverse().join("");
console.log(textInvertMethod); // Resultado: "odnuM aloH"

let textInvertFor = "";
for (let i = cadena.length - 1; i >= 0; i--) {
  textInvertFor += cadena[i];
  console.log(textInvertFor);
}
console.log(textInvertFor); // Resultado: "odnuM aloH"
